napari/components/_layers/view/main.py

```diff
@@ -92,10 +151,11 @@ def dragMoveEvent(self, event):
         center_list = (i for i, x in enumerate(self.centers) if x > cord)
         divider_index = next(center_list, len(self.centers))
         # Determine the current location of the widget being dragged
-        layerWidget = event.source()
         total = self.vbox_layout.count()//2 - 1
-        index = total - self.vbox_layout.indexOf(layerWidget)//2 - 1
         insert = total - divider_index
+        layer_name = event.mimeData().text()
+        layer = self.layers[layer_name]
```

Comment:
if you don't need `layer` you can just get the index directly from the layer name

Reply:
fixed